rebuild/src/addon.cc

```diff
@@ -0,0 +1,85 @@
+#include "addon.h"
+
+bool is_zero_bytes(const uint8_t *data,
+                   const size_t start_byte,
+                   const size_t byte_length)
+{
+    for (size_t i = start_byte; i < byte_length; i++)
+    {
+        if (data[i] != 0)
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool is_valid_length(
+    std::string &error_out,
+    std::string_view error_prefix,
```

Comment:
can we put them in some errCtx structure, having two fields to manager error is a bit much

Reply:
I removed `error_prefix` and set that to the output `error_out` before passing the reference. I also added the in/out to @param defs